app/org/sagebionetworks/bridge/services/AuthenticationServiceImpl.java

```diff
@@ -353,11 +191,12 @@ public UserSession getSessionFromAccount(Study study, Account account) {
         user.setStudyKey(study.getIdentifier());
 
         final String healthCode = getHealthCode(study, account);
-        user.setHealthCode(healthCode);
-
-        user.setSignedMostRecentConsent(consentService.hasUserSignedMostRecentConsent(user, study));
-        user.setConsent(consentService.hasUserConsentedToResearch(user, study));
-        user.setDataSharing(optionsService.getBooleanOption(healthCode, Option.DATA_SHARING));
+        if (healthCode != null) {
+            user.setHealthCode(healthCode);
+            user.setDataSharing(optionsService.getBooleanOption(healthCode, Option.DATA_SHARING));
+        }
```

Comment:
[POLISH] Under what conditions would this would healthCode be null? getHealthCode() creates the HealthId if it's not present, and HealthCodeService.createMapping() ensures both are created. Plus, the semantics of getSession() is that it will create the health code if not present, and this code looks confusingly like the healthcode might not have been created at this point.